multiverse-source/src/lib.rs

```diff
@@ -0,0 +1,339 @@
+mod point;
+
+use anyhow::Result;
+use dcspark_blockchain_source::{EventObject, PullFrom, Source};
+use multiverse::{BestBlock, BestBlockSelectionRule, Variant};
+pub use point::Point;
+use std::{
+    fmt::{Debug, Display},
+    hash::Hash,
+};
+
+pub struct Multiverse<K, V, InnerSource> {
+    multiverse: multiverse::Multiverse<K, V>,
```

Comment:
do we need a separate crate for this by the way? this is still a source, we have a crate for them

Reply:
not really, but I'm a bit confused about the meaning of *blockchain*_source, since I'm not sure I would qualify this that way. I took it quite literal, to mean things like cardano/algorand/etc. A separate crate could lead to slight improvements in compile time in certain cases, but this is quite small, so I highly doubt it makes any difference. I'll merge with the other one.